neuro_san/service/http/handlers/base_request_handler.py

```diff
@@ -65,7 +64,12 @@ def initialize(self,
         if os.environ.get("AGENT_ALLOW_CORS_HEADERS") is not None:
             self.set_header("Access-Control-Allow-Origin", "*")
             self.set_header("Access-Control-Allow-Methods", "GET, POST, OPTIONS")
-            self.set_header("Access-Control-Allow-Headers", "Content-Type, Transfer-Encoding, User_id")
+            headers: str = "Content-Type, Transfer-Encoding"
+            metadata_headers: str = ", ".join(forwarded_request_metadata)
+            if len(metadata_headers) > 0:
+                headers += f", {metadata_headers}"
+            # Set all allowed headers:
+            self.set_header("Access-Control-Allow-Headers", headers)
 
```

Comment:
Generate CORS headers list from "forwarded_request_metadata" collection passed in.